pennylane/devices/experimental/device_api.py

```diff
@@ -87,6 +87,28 @@ class Device(abc.ABC):
         Versioning should be specified by the package containing the device. If an external package includes a PennyLane device,
         then the package requirements should specify the minimium PennyLane version required to work with the device.
 
+    .. details::
+        :title: The relationship between preprocessing and execution
+
+        The :meth:`~.preprocess` method is assumed to be run before any :meth:`~.execute` or differentiation method.
+        If an arbitrary, non-preprocessed circuit is provided, :meth:`~.execute` has no responsibility to perform any
+        validation or provide clearer error messages.
```

Comment:
There is a bit of validation related to multiprocessing in `execute` right now. So is that ok because while we can't assume it, `execute` might still do some runtime validation, or should we move that bit to `preprocess`?